tfe/resource_tfe_policy_set.go

```diff
@@ -275,3 +355,36 @@ func resourceTFEPolicySetDelete(d *schema.ResourceData, meta interface{}) error
 
 	return nil
 }
+
+func validatePolicySetSourceTypes(d *schema.ResourceDiff, meta interface{}) error {
+	_, vcsOk := d.GetOk("vcs_repo")
+	_, pathOk := d.GetOk("policies_path")
+	_, policiesOk := d.GetOk("policy_ids")
+
+	// TODO: The list/set validations here use the direct count keys
+	// and may need to be changed once the Terraform SDK is better set
+	// up to support TF 0.12 idioms.
+	//
+	// Note that it's possible that this is a bug in the SDK too - you
+	// should definitely be able to check the root-level key for a
+	// block/list/set to see if it's unknown. Not too sure why that's
+	// not currently happening.
+	vcsOk = vcsOk || !d.NewValueKnown("vcs_repo.#")
+	pathOk = pathOk || !d.NewValueKnown("policies_path")
+	policiesOk = policiesOk || !d.NewValueKnown("policy_ids.#")
+
+	// Ensure that at least one of vcs_repo or policy_ids defined. We
+	// don't need to check for conflicts as that is handled in
+	// validation.
+	if !vcsOk && !policiesOk {
+		return errors.New("one of vcs_repo or policy_ids must be defined")
+	}
+
+	// Ensure that policies_path is *not* supplied when vcs_repo is
+	// undefined.
+	if !vcsOk && pathOk {
+		return errors.New("policies_path must not be supplied when vcs_repo is empty")
+	}
+
+	return nil
+}
```

Comment:
Can you explain why we need this custom logic here? Is there no way we can achieve the same using by configuring the schema correctly?

Reply:
No, we don't allow mutual inclusivity in the SDK right now individual validations only operate at the individual attribute level, without the ability to access other fields.

Reply:
I'm kind of wondering if, based on the outcome of https://github.com/hashicorp/go-tfe/pull/77#issuecomment-502185845, if we even need any of these checks. We could always just have the resource default to creating an empty set without VCS if none of the fields are specified. I can also test to see what happens if you do specify a `policies_path` field without VCS repo options - the docs imply that it fails, but if it ignores it, we can just do that too. We could even just omit it from the API request for legacy sets on our side anyway and just remove this code, and indicate in the docs that it's ignored.

Reply:
I've gone and removed the default. This gets rid of the need for the validation during diff customization.

I've re-purposed one of the tests to make sure that it does the expected default behavior, in addition to another one that validates that going from a populated set to an empty set via default behavior is possible.